snap/info.go

```diff
@@ -547,6 +547,8 @@ type PlugInfo struct {
 	Label     string
 	Apps      map[string]*AppInfo
 	Hooks     map[string]*HookInfo
+
+	TopLevel bool
```

Comment:
I'm wondering whether toplevelSlots and toplevelPlugs lists in Info wouldn't serve us better.

My concernt is that this kind of infos are used all over the place, and it's a strange concern to decide whether they are top level or not.